site/content/entry/yuzu-progress-report-may-2021/index.md

```diff
@@ -0,0 +1,350 @@
+motion input from valid controllers like Dual Sense, Joy-Cons, or Pro Controllers.
+With this PR, users can simply pair their controllers via Bluetooth and yuzu will pick them up as a valid input device, auto-mapping and enabling motion support, effectively 
+removing the need to use betterjoy, ds4windows, or other third party software in yuzu.
+
+A separate PR adds [detection of two separate Joy-Cons as a single input device](https://github.com/yuzu-emu/yuzu/pull/6318).
+
+And to provide motion support on the Linux side, [v1993](https://github.com/v1993) [enables the use of HIDAPI within SDL2](https://github.com/yuzu-emu/yuzu/pull/6293). 
+
+Work continues to fine tune how the sensor data is interpreted, specially from non-Nintendo devices.
+For example, PS3 controllers operate motion on a single Z axis, but report the remaining 2 axes to the operating system with useless NaN values, 
+making any motion calculations useless.
+[By replacing the X and Y axis values with zeros](https://github.com/yuzu-emu/yuzu/pull/6310), german77 solved the issue.
+
+Games don’t always check for obvious physical limitations. 
+For example, in the past, it was possible to use handheld controller emulation while the game was in a docked status.
+Doing this can break games like `Xenoblade Chronicles 2`, as the game expects the console to disable input when the Joy-Cons are connected to a docked Switch’s rails.
+german77 solves this by [forcing an emulated Pro Controller instead](https://github.com/yuzu-emu/yuzu/pull/6353). 
+No more weird Frankenstein input monstrosities, please.
+
+Finally, it could sometimes be difficult to map analog sticks in yuzu due to a bug in how the event was handled. While fixing this, german77 also 
+[fixed a crash caused by controllers not listed in the controller list disconnecting](https://github.com/yuzu-emu/yuzu/pull/6312).
+
+## Future projects
+
+Now, I know we keep teasing `Project Hades`, but we have good news!
+SPIR-V is finalized, GLASM is in its final testing, and GLSL is not far behind. Few regressions remain to be squashed.
+Once the shading languages are done, previous features like the old asynchronous shaders are reimplemented, and the fine tuning is finished, it will finally be released.
+We're trying to release it alongside Reaper, but time will tell.
+
+{{< single-title-imgs
+    "MARVEL ULTIMATE ALLIANCE 3: The Black Order, WORLD OF FINAL FANTASY MAXIMA, Yoshi's Crafted World"
+    "./hades1.png"
+    "./hades2.png"
+    "./hades3.png"
+  >}}
+
+That’s all folks! Thank you so much for staying with us until the end of this month’s progress report.
+See you in the next one!
+
+&nbsp;
+<h4 style="text-align:center;">
+<b>Please consider supporting us on [Patreon](https://www.patreon.com/yuzuteam)!<br>
+If you would like to contribute to this project, check out our [GitHub](https://github.com/yuzu-emu/yuzu)!</b>
+</h4>
+
```

Comment:
With new Hugo update, replace this ending HTML with the following shortcode:

`{{< article-end >}}`